File: tests/uasm/sample/licm/LicmTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import type { UdonInt } from "@ootr/udon-assembly-ts/stubs/UdonTypes";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class LicmTest extends UdonSharpBehaviour {
  Start(): void {
    const baseVal: UdonInt = 10n as UdonInt;
    const mult: UdonInt = 3n as UdonInt;
    let sum: UdonInt = 0n as UdonInt;
    for (let i: UdonInt = 0n as UdonInt; i < (5n as UdonInt); i++) {
      const invariant: UdonInt = (baseVal * mult) as UdonInt;
      sum = (sum + invariant) as UdonInt;
    }
    Debug.Log(sum);
  }
}

File: tests/uasm/sample/licm/LicmTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class LicmTest : UdonSharpBehaviour
{
    public void Start()
    {
        int baseVal = 10;
        int mult = 3;
        int sum = 0;
        for (int i = 0; i < 5; i++)
        {
            int invariant = baseVal * mult;
            sum = sum + invariant;
        }
        Debug.Log(sum);
    }
}
